tests/functions/test_fill.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+df = pd.DataFrame(
+    [
+        {
+            "rank": 1,
+            "pet_type": np.nan,
+            "breed": "Boston Terrier",
+            "owner": "sam",
+        },
+        {
+            "rank": 2,
+            "pet_type": np.nan,
+            "breed": "Retrievers (Labrador)",
+            "owner": "ogor",
+        },
+        {
+            "rank": 3,
+            "pet_type": np.nan,
+            "breed": "Retrievers (Golden)",
+            "owner": "nathan",
+        },
+        {
+            "rank": 4,
+            "pet_type": np.nan,
+            "breed": "French Bulldogs",
+            "owner": np.nan,
+        },
+        {"rank": 5, "pet_type": np.nan, "breed": "Bulldogs", "owner": np.nan},
+        {"rank": 6, "pet_type": "Dog", "breed": "Beagles", "owner": np.nan},
+        {"rank": 1, "pet_type": np.nan, "breed": "Persian", "owner": np.nan},
+        {
+            "rank": 2,
+            "pet_type": np.nan,
+            "breed": "Maine Coon",
+            "owner": "ragnar",
+        },
+        {"rank": 3, "pet_type": np.nan, "breed": "Ragdoll", "owner": np.nan},
+        {"rank": 4, "pet_type": np.nan, "breed": "Exotic", "owner": np.nan},
+        {"rank": 5, "pet_type": np.nan, "breed": "Siamese", "owner": np.nan},
+        {
+            "rank": 6,
+            "pet_type": "Cat",
+            "breed": "American Short",
+            "owner": "adaora",
+        },
+    ]
+)
```

Comment:
For maintenance (and cleanliness in the test), can we make this one a fixture? I think the way to do this is:

```suggestion
@pytest.fixture
def fill_dummy_data():
    df = pd.DataFrame(
        [
            {
                "rank": 1,
                "pet_type": np.nan,
                "breed": "Boston Terrier",
                "owner": "sam",
            },
            {
                "rank": 2,
                "pet_type": np.nan,
                "breed": "Retrievers (Labrador)",
                "owner": "ogor",
            },
            {
                "rank": 3,
                "pet_type": np.nan,
                "breed": "Retrievers (Golden)",
                "owner": "nathan",
            },
            {
                "rank": 4,
                "pet_type": np.nan,
                "breed": "French Bulldogs",
                "owner": np.nan,
            },
            {"rank": 5, "pet_type": np.nan, "breed": "Bulldogs", "owner": np.nan},
            {"rank": 6, "pet_type": "Dog", "breed": "Beagles", "owner": np.nan},
            {"rank": 1, "pet_type": np.nan, "breed": "Persian", "owner": np.nan},
            {
                "rank": 2,
                "pet_type": np.nan,
                "breed": "Maine Coon",
                "owner": "ragnar",
            },
            {"rank": 3, "pet_type": np.nan, "breed": "Ragdoll", "owner": np.nan},
            {"rank": 4, "pet_type": np.nan, "breed": "Exotic", "owner": np.nan},
            {"rank": 5, "pet_type": np.nan, "breed": "Siamese", "owner": np.nan},
            {
                "rank": 6,
                "pet_type": "Cat",
                "breed": "American Short",
                "owner": "adaora",
            },
        ]
    )
    return df
```

And then in each of the tests, you would pass in `fill_dummy_data` as the argument, as follows:

```python
def test_fill_column(fill_dummy_data):
    ...
```

`pytest` will automagically pass in the result of `fill_dummy_data`. Doing the test as a fixture makes the dataframe safe from global mutation (in case it might ever happen), so we don't accidentally change the structure of the dataframe in memory when using it in a test that executes after the mutation.

Reply:
I'll be leaving "fixture please" comments (copy/pasted) so you can have a pseudo-todo list on the PR tracker.